app/ydoc-server/src/fileFormat.ts

```diff
@@ -38,8 +38,8 @@ export type IdeMetadata = z.infer<typeof ideMetadata>
 export const ideMetadata = z
   .object({
     node: z.record(z.string().uuid(), nodeMetadata),
-    snapshot: z.string().optional(),
     widget: z.optional(z.record(z.string().uuid(), z.record(z.string(), z.unknown()))),
+    snapshot: z.string().optional(),
```

Comment:
Make the `snapshot` the last field. The ydoc diff algorithm places the `snapshot` at the end of the metadata. This prevents unnecessary edits.

Reply:
Oh, that's subtle--it could use a comment.